core/trino-main/src/main/java/io/trino/sql/planner/optimizations/TransformQuantifiedComparisonApplyToCorrelatedJoin.java

```diff
@@ -253,6 +253,8 @@ private static boolean shouldCompareValueWithLowerBound(QuantifiedComparisonExpr
                         case GREATER_THAN:
                         case GREATER_THAN_OR_EQUAL:
                             return false;
+                        default:
+                            // Caller guarantees no other cases need to be handled here
```

Comment:
does this mean we should throw here?

Reply:
we throw at the end of the method, so i decided not to throw here.

Generally the `MissingCasesInEnumSwitch` is most useful when you want to have _all options_ covered, and not useful when you consciously cover only some of them.
However, this is not visible in this PR -- someone should go over the code and check which `default:` clauses can be _removed_, in order to make the code tap into `MissingCasesInEnumSwitch`-provided security.